tkg/client/audit_upgrade_helper.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"strings"
+
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+)
+
+const (
+	auditConfigFilePath = "/etc/kubernetes/audit-policy.yaml"
+	apiVersionV1Alpha1  = "apiVersion: audit.k8s.io/v1alpha1"
+	apiVersionV1Beta1   = "apiVersion: audit.k8s.io/v1beta1"
+	apiVersionV1        = "apiVersion: audit.k8s.io/v1"
+)
+
+func (c *TkgClient) configureAuditVersion(old *controlplanev1.KubeadmControlPlane) *controlplanev1.KubeadmControlPlane {
+	for i, f := range old.Spec.KubeadmConfigSpec.Files {
+		if f.Path == auditConfigFilePath {
+			if strings.Contains(f.Content, apiVersionV1Alpha1) || strings.Contains(f.Content, apiVersionV1Beta1) {
+				kcp := old.DeepCopy()
+				newFile := f.DeepCopy()
+				newContent := strings.ReplaceAll(f.Content, apiVersionV1Beta1, apiVersionV1)
+				newFile.Content = strings.ReplaceAll(newContent, apiVersionV1Alpha1, apiVersionV1)
```

Comment:
Got a little confused here, why the ReplaceAll called twice instead of using below directly?
```
newFile.Content = newContent
```